javascript-modules-library/README.md

```diff
@@ -1,36 +1,212 @@
-repository as the scripts will automatically do that for you but we mention it in case you need to make modifications
-to it.
+### `RenderInBrowser`
 
-By default, the generated types are committed in this project, but if you need to rebuild/update them, you can do so by
-using the following steps:
+This component is used to render a React component in the browser. It is used to render a component in the browser, skipping the server-side rendering step. Provided children, if any, will be rendered on the server and replaced by the component in the browser.
 
-1.  Change to the project directory :
+```tsx
+<RenderInBrowser child={MyComponent} props={{ foo: "bar" }}>
+  <div>Loading...</div>
+</RenderInBrowser>
+```
 
-    `cd types/generated`
+### `HydrateInBrowser`
 
-2.  Launch the script that will clone the java-ts-bind repository and build it from source code (it has never been released) :
+This component is used to hydrate a React component in the browser. It will be rendered on the server then hydrated in the browser. Provided children, if any, will be children of the component.
 
-    `./build.sh`
+```tsx
+<HydrateInBrowser child={MyComponent} props={{ foo: "bar" }}>
+  <div>I'm a child of MyComponent</div>
+</HydrateInBrowser>
+```
 
-3.  Update the classes and methods declared in [types/generated/package.json](types/generated/package.json) in case you added/removed classes or methods.
+### `Render`
 
-4.  Launch the script that will download all the required source code and JARs for the type generation and that will
-    also compile the source to generated the required symbols JARs. Then it will generate the types, and finally apply
-    some post-processing to the generated types to make usable :
+This component renders a Jahia component out of a node or a JS object.
 
-        `./generate.sh`
+```tsx
+// Render a JCR node
+<Render node={node} />
 
-Note that the generation will take some time on the initial build as it needs to download all the required source code
-and JARs, and build the projects from the source code.
+// Render a JS object
+<Render content={{ nodeType: "ns:nodeType" }}}>
+```
 
-Once completed, the generated types will be in the following directory :
+## Components
 
-    types/generated/build/types
+### `AbsoluteArea`
+
+This component creates an absolute area in the page. It's an area of user-contributable content that is synchronized between all pages where it is present.
+
+```tsx
+<AbsoluteArea name="footer" />
+```
+
+### `Area`
+
+This component creates an area in the page. It's an area of user-contributable content that is local to the page where it is present.
+
+```tsx
+<Area name="main" />
+```
+
+### `AddContentButtons`
+
+This component renders a set of buttons to add content to the current node.
+
+```tsx
+<AddContentButtons />
+```
+
+### `AddResources`
+
+This component adds resources to the page, making sure they are loaded only once and insert them at the desired position.
```

Comment:
Should we clarify that "desired position" means in the `<head>` of the page?

Reply:
This is documented in more details in the JSDoc of the component, I'd say we should not make this summary longer than it is now